### PR TITLE
SoundCloud expando #216

### PR DESCRIPTION
**Pull request by kiwwisk**
Implementing SoundCloud expando (No API key necessary).

soundcloud_expando

### CODE REVIEW

**Comment by kiwwisk** [comment]
This is pull request for implementing Issue #216

**Comment by PuttItOut** [comment]
Thanks for this PR. I didn't know it was possible to access soundcloud without an API key. I'll test this once everything calms down.

**Comment by kiwwisk** [comment]
Using OEmbed protocol it is possible to retrieve all embedding information from SoundCloud when necessary. It also opens the path for supporting different sites with oembed in the future.

File: Whoaverse/Whoaverse/Scripts/whoaverse.ui.js [hunk 1]
```diff
@@ -809,15 +820,44 @@ VimeoExpando.prototype.process = function (target) {
...
+            $.getJSON(UI.Common.currentProtocol() + '//soundcloud.com/oembed', {
+                format: 'json',
+                url: 'https://soundcloud.com/' + id,
+                iframe: 'true'
+            }).done(function (json) {
+                if (!json.html) {
+                    fnErrorHandler();
+                    return;
+                }
+                var regex = /<iframe.*?src="(.*?)"/;
```
Comment (Romoku) [inline Whoaverse/Whoaverse/Scripts/whoaverse.ui.js]: I think it is useful to comment regular expressions for people who are not familiar.

Comment (kiwwisk) [inline Whoaverse/Whoaverse/Scripts/whoaverse.ui.js]: Added a regex comment to my pull request.

**Comment by voat** [comment]
Thank you for this PR. This can no longer be automerged due to recent whoaverse->voat rename. Would you mind updating your fork and this PR?

**Comment by PuttItOut** [comment]
Closing because of conflicts and no resubmittal. Thanks a ton for your effort.

**Pull request closed without merging**